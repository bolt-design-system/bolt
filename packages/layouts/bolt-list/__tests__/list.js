import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
  vrtDefaultConfig,
} from '../../../testing/testing-helpers';
import schema from '../list.schema';
const {
  display,
  spacing,
  separator,
  inset,
  align,
  valign,
  tag,
  nowrap,
} = schema.properties;

const timeout = 120000;

describe('<bolt-list> Component', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic usage', async () => {
    const results = await render('@bolt-layouts-list/list.twig', {
      items: ['item 1', 'item 2', 'item 3'],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
