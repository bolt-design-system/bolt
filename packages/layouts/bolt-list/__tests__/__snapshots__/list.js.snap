// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-list> Component basic usage 1`] = `
<bolt-list tag="ul"
           display="block"
           spacing="xsmall"
           separator="none"
           align="start"
           valign="center"
>
  <ssr-keep for="bolt-list"
            role="list"
            class="c-bolt-list c-bolt-list--display-block c-bolt-list--spacing-xsmall c-bolt-list--align-start c-bolt-list--valign-center"
  >
    <bolt-list-item role="presentation">
      <ssr-keep for="bolt-list-item"
                role="listitem"
                class="c-bolt-list-item c-bolt-list-item--display-block c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start"
      >
        item 1
      </ssr-keep>
    </bolt-list-item>
    <bolt-list-item role="presentation">
      <ssr-keep for="bolt-list-item"
                role="listitem"
                class="c-bolt-list-item c-bolt-list-item--display-block c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start"
      >
        item 2
      </ssr-keep>
    </bolt-list-item>
    <bolt-list-item last
                    role="presentation"
    >
      <ssr-keep for="bolt-list-item"
                role="listitem"
                class="c-bolt-list-item c-bolt-list-item--display-block c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start c-bolt-list-item--last-item"
      >
        item 3
      </ssr-keep>
    </bolt-list-item>
  </ssr-keep>
</bolt-list>
`;
